service/src/main/java/io/camunda/service/RoleServices.java

```diff
@@ -52,24 +53,25 @@ public RoleServices withAuthentication(final Authentication authentication) {
         brokerClient, securityContextProvider, roleSearchClient, authentication);
   }
 
-  public CompletableFuture<RoleRecord> createRole(final RoleDTO request) {
-    return sendBrokerRequest(new BrokerRoleCreateRequest().setName(request.name()));
+  public CompletableFuture<RoleRecord> createRole(final String name) {
+    return sendBrokerRequest(new BrokerRoleCreateRequest().setName(name));
   }
 
-  public CompletableFuture<RoleRecord> updateRole(final RoleDTO request) {
-    return sendBrokerRequest(
-        new BrokerRoleUpdateRequest(request.roleKey()).setName(request.name()));
+  public CompletableFuture<RoleRecord> updateRole(final long roleKey, final String name) {
+    return sendBrokerRequest(new BrokerRoleUpdateRequest(roleKey).setName(name));
   }
 
-  public RoleEntity getByRoleKey(final Long roleKey) {
-    final SearchQueryResult<RoleEntity> result =
-        search(SearchQueryBuilders.roleSearchQuery().filter(f -> f.roleKey(roleKey)).build());
-    if (result.total() < 1) {
-      throw new NotFoundException(String.format("Role with roleKey %d not found", roleKey));
-    } else {
-      return result.items().stream().findFirst().orElseThrow();
-    }
+  public RoleEntity getRole(final Long roleKey) {
+    return findRole(roleKey)
+        .orElseThrow(
+            () -> new NotFoundException("Role with roleKey %d not found".formatted(roleKey)));
   }
 
-  public record RoleDTO(long roleKey, String name, Set<Long> assignedMemberKeys) {}
+  @VisibleForTesting
+  public Optional<RoleEntity> findRole(final Long roleKey) {
```

Comment:
🔧 As far as I'm aware, `VisibleForTesting` is not meant for public members. The Javadoc of this annotation says
> Indicates that the visibility was strengthened purely for testing purposes.
> NOTE: this should not be used on public members, but is meant only to highlight package-private or protected fields, types, etc., which aren't private in order to allow comprehensive tests. In general, you should try to avoid this, but at times there is no other way.